cmd/zqd/logger/logger.go

```diff
@@ -1,35 +1,83 @@
 package logger
 
 import (
+	"fmt"
+	"io/ioutil"
+
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
+type Type string
+
+const (
+	TypeFile      Type = "file"
+	TypeWaterfall      = "waterfall"
+	TypeTee            = "tee"
+)
+
 type Config struct {
 	Path string `yaml:"path"`
 	// If Path is a file, Mode will determine how the log file is managed.
 	// FileModeAppend is the default if value is undefined.
 	Mode  FileMode      `yaml:"mode,omitempty"`
 	Name  string        `yaml:"name"`
 	Level zapcore.Level `yaml:"level"`
+	// Type specifies how log entries are handled. If Type is Waterfall or Tee
```

Comment:
I see that this looks designed to make the existing Brim default logger config continue to work:
https://github.com/brimsec/brim/blob/master/src/js/zqd/zqd.js#L35

Did you try it out to make sure its ok? If so, once we cut a zqd release with this PR's change & update Brim to use it, should we change that default config in Brim?

Reply:
The config file in brim will have to change. It should look like this instead:
```yaml
logger:
  type: waterfall
  children:
  - name: http.access
     level: info
     path: ${accessLogFile}
     mode: rotate
  - level: info
     path: ${zqdLogFile}
     mode: rotate
```

Reply:
I have not checked if this works with Brim. Doing that now.

Reply:
Verified that if you make the above changes, this will work as desired with brim.